### PR TITLE
Shortcut for using agents from the projects folder

### PR DESCRIPTION
**Pull request by emilydinan**
Similar to the way we use `internal:` to get agents in internal, look for agents in a projects folder by specifying `projects:` and the name of the projects folder.

This is useful for using `projects:personachat:kvmemnn` or for when we add project specific model code (like we are doing with wizard).

### CODE REVIEW

File: parlai/core/agents.py [hunk 1]
```diff
@@ -392,6 +396,18 @@ def get_agent_module(dir_name):
...
+    elif dir_name.startswith('projects:'):
+        # e.g. -m projects:personachat:kvmemnn
+        s = dir_name.split(':')
+        if len(s) != 3:
+            raise RuntimeError('projects paths should follow pattern '
+                               'projects:folder:model; you used {}'
+                               ''.format(dir_name))
+        folder_name = s[1]
+        model_name = s[2]
+        module_name = 'projects.{v}.{m}.{m}'.format(
```
Comment (stephenroller) [inline parlai/core/agents.py]: let's make it `{p}` instead of `{v}`.

File: parlai/core/agents.py [hunk 2]
```diff
@@ -392,6 +396,18 @@ def get_agent_module(dir_name):
         module_name = 'parlai.agents.legacy_agents.{m}.{m}_v{v}'.format(
             m=model_name, v=s[2])
         class_name = name_to_agent_class(model_name)
+    elif dir_name.startswith('projects:'):
+        # e.g. -m projects:personachat:kvmemnn
+        s = dir_name.split(':')
+        if len(s) != 3:
+            raise RuntimeError('projects paths should follow pattern '
+                               'projects:folder:model; you used {}'
+                               ''.format(dir_name))
```
Comment (stephenroller) [inline parlai/core/agents.py]: Maybe we should say that this maps to `parlai.projects.folder.model.model`?

File: parlai/core/agents.py (hunk 1 above)
Comment (stephenroller) [reply parlai/core/agents.py]: Just wondering: does this automatically insert the `parlai.`? It probably should?

Also wondering how this deals with capitalizations

File: parlai/core/agents.py (hunk 2 above)
Comment (emilydinan) [reply parlai/core/agents.py]: it doesn't, projects is outside of parlai

**Comment by stephenroller** [comment]
Feel free to merge without travis because this didn't incorporate #1407

**Pull request merged**